jsonschema/schema.go

```diff
@@ -167,3 +167,40 @@ func mergeProperties(parent, child map[string]json.RawMessage) map[string]json.R
 	}
 	return newProperties
 }
+
+// FieldType returns the type name of a field and whether the field is nullable.
+func FieldType(data []byte) (string, bool, error) {
+	// The "type" field in JSON Schema is either a scalar or an array.
+	value, dataType, _, err := jsonparser.Get(data, "type")
+	if err != nil && !errors.Is(err, jsonparser.KeyPathNotFoundError) {
+		return "", false, fmt.Errorf("error reading schema type: %w", err)
+	} else if dataType == jsonparser.String {
+		return string(value), false, nil
+	} else if dataType == jsonparser.Array {
+		var nullable bool
+		var jsonType string
+		_, aErr := jsonparser.ArrayEach(value, func(avalue []byte, adataType jsonparser.ValueType, aoffset int, aerr error) {
+			if adataType != jsonparser.String {
+				return
+			}
+			switch string(avalue) {
+			case "null":
+				nullable = true
+			default:
+				if jsonType != "" {
+					err = fmt.Errorf("union types are not supported: %s", value)
+				}
```

Comment:
Just confirming that you don't want to just return the err here?

Reply:
Can't return an error there because the `jsonparser.ArrayEach()` calls that as a sub-function, hence the odd dance after it.